### PR TITLE
Added printing of captured stdout before entering pdb

### LINKED ISSUE #3052: Possible regression of pdb suppressing output
Hey guys, /cc @nicoddemus

Approx two years ago I submitted a patch for `--pdb` to stop suppressing output, seen here;
https://github.com/pytest-dev/pytest/issues/1223

I can see the PR was merged and the appropriate test is present in master;
https://github.com/pytest-dev/pytest/blob/5c6d7739bcd7a1ea33eecff37bd2c2271cf5309c/testing/test_pdb.py#L150

But when trying to use this myself, it doesn't seem to work, as seen below.

Any thoughts why this would pass unit tests, but wouldn't work in latest release?

```
user@c5595a22fd91:~/src/wtf$ cat test_all.py
def test_all():
    print('getrekt')
    assert False
```

```
user@c5595a22fd91:~/src/wtf$ pytest --pdb test_all.py
============================================================================================================ test session starts =============================================================================================================
platform linux -- Python 3.6.3, pytest-3.3.1, py-1.5.2, pluggy-0.6.0 -- /usr/bin/python3
cachedir: ../.cache
rootdir: /home/user/src, inifile: pytest.ini
collected 1 item
run-last-failure: run all (no recorded failures)

test_all.py::test_all FAILED                                                                                                                                                                                                           [100%]
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> traceback >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
Traceback (most recent call last):
  File "/home/user/src/wtf/test_all.py", line 3, in test_all
    assert False
AssertionError: assert False
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> entering PDB >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
> /home/user/src/wtf/test_all.py(3)test_all()
-> assert False
```

### PR DESCRIPTION
**Pull request by brianmaissy**
Fixes #3052

### CODE REVIEW

File: _pytest/debugging.py [hunk 1]
```diff
@@ -85,6 +85,8 @@ def _enter_pdb(node, excinfo, rep):
     # for not completely clear reasons.
     tw = node.config.pluginmanager.getplugin("terminalreporter")._tw
     tw.line()
+    tw.sep(">", "captured stdout")
```
Comment (nicoddemus) [inline _pytest/debugging.py]: Perhaps we should print this only if there's any actual stdout:

```python
stdout = rep.capstdout
if stdout:
    tw.sep(">", "captured stdout")
    tw.line(stdout)
```

And maybe we should also print `stderr`?

Comment (brianmaissy) [reply _pytest/debugging.py]: both sound good

**Comment by nicoddemus** [comment]
@brianmaissy can you take a look at the failing test please?

File: _pytest/debugging.py [hunk 2]
```diff
@@ -85,6 +85,17 @@ def _enter_pdb(node, excinfo, rep):
...
+    captured_stdout = rep.capstdout
+    if len(captured_stdout) > 0:
+        tw.sep(">", "captured stdout")
+        tw.line(captured_stdout)
+
+    captured_stderr = rep.capstderr
+    if len(captured_stderr) > 0:
+        tw.sep(">", "captured stderr")
+        tw.line(captured_stderr)
+
```
Comment (thisch) [inline _pytest/debugging.py]: Please also output the captured logs (``rep.caplog``) and make sure that they are not output if live-logging is enabled (``-o log_cli=True -o log_level=DEBUG``)

Comment (thisch) [reply _pytest/debugging.py]: Note that ``rep.caplog`` was added in the features branch. So you can't use it if you merge your branch into master, but you can copy'n'paste the implemenation of BaseReport.caplog from https://github.com/pytest-dev/pytest/pull/3156/files#diff-e1a5c52f502faf325466018a5440c400R260.

Comment (nicoddemus) [reply _pytest/debugging.py]: > Please also output the captured logs (rep.caplog) and make sure that they are not output if live-logging is enabled (-o log_cli=True -o log_level=DEBUG)

Why should we make a special case for living log here? I kind see the point, but we are dumping captured stdout/stderr at this point it follows that we should dump caplog then as well, regardless of live logging. 

> but you can copy'n'paste the implemenation of BaseReport.caplog from

If we end up doing this, please don't make `caplog` public: we don't want to introduce a new public API in a bug-fix release.

Comment (brianmaissy) [reply _pytest/debugging.py]: maybe it would be better to merge this and open a new issue to implement the same thing for caplog, in the features branch

Comment (nicoddemus) [reply _pytest/debugging.py]: @brianmaissy fair enough. @thisch do you agree?

Comment (thisch) [reply _pytest/debugging.py]: Ok. Let's do it like this

Comment (brianmaissy) [reply _pytest/debugging.py]: Opened a new issue #3204

**Comment by brianmaissy** [comment]
finally got that failing test

**Comment by thisch** [comment]
Great! I think you can squash your last three commits into one. (IMO you can squash all your commits into one. @nicoddemus, do you agree?)

**Comment by nicoddemus** [comment]
Thanks @brianmaissy!

**Comment by thisch** [comment]
Thx again!

**Pull request merged**